proxy/executor/strategy_base.py

```diff
@@ -255,16 +272,33 @@ async def _init_sol_tx_cfg(
         ix_mode: NeonIxMode = NeonIxMode.Unknown,
         cu_limit: int = 0,
         cu_price: int = 0,
-        heap_size: int = 0
+        heap_size: int = 0,
     ) -> SolTxCfg:
+        # TODO EIP1559 churn: remove atlas.
+        # if not cu_price:
+        #    cu_price = await self._ctx.fee_client.get_cu_price(self._ctx.rw_account_key_list)
+        cu_limit = cu_limit or self._cu_limit
         if not cu_price:
-            cu_price = await self._ctx.fee_client.get_cu_price(self._ctx.rw_account_key_list)
+            if self._ctx.tx_type == 0:
+                # For legacy transactions: we estimate the cu_price from the recent blocks.
+                cu_price = await self._estimate_cu_price()
+            else:
+                base_fee_per_gas = self._ctx.max_fee_per_gas - self._ctx.max_priority_fee_per_gas
+                assert base_fee_per_gas >= 0
+                # For metamask case (max_fee_per_gas = max_priority_fee_per_gas), we treat it as a legacy transaction.
+                if base_fee_per_gas == 0:
+                    cu_price = await self._estimate_cu_price()
+                else:
+                    # For general Dynamic Gas transactions: we take the cu_price according to what the User set in Neon tx.
+                    cu_price = max(
+                        1, int(self._ctx.max_priority_fee_per_gas * 1_000_000 * 5000.0 / (base_fee_per_gas * cu_limit))
```

Comment:
EIP-1559 is about SAVING money on PRIORITY
It isn't about FORCING the PRIORITY by a user.

The rule should be
- User sets the MAX priority fee (NeonTx.priority_fee)
- Proxy calculates the CURRENT priority fee in network (network.priority_fee)
- Proxy sets the SolanaTx.priority_fee = min(NeonTx.priority_fee, network.priority_fee)
- EVM  pays to operator min(NeonTx.priority_fee, SolanaTx.priority_fee)

It means that EVM protects the User from the Operator desire to get more money on priorities.

Reply:
```
   cu_price = min(
      await self._estimate_cu_price(),
      max(
         1, 
         int(self._ctx.max_priority_fee_per_gas * 1_000_000 * 5000.0 / (base_fee_per_gas * cu_limit)
      )
   )
 ```